src/FsCheck/StateMachine.fs

```diff
@@ -246,22 +246,33 @@ module StateMachine =
 
     [<CompiledName("Generate")>]
     let generate (spec:Machine<'Actual,'Model>) = 
-        let rec genCommandsS state size =
+        let retryNumber = 100
+        let rec genCommandsS state size retries =
             gen {
                 if size > 0 then
                     let nextOperation = spec.Next state
                     let! command = nextOperation |> Gen.tryWhere (fun operation -> operation.Pre state)
-                    if Option.isNone command || command.Value.GetType() = typeof<StopOperation<'Actual,'Model>> then return [state],[]
+                    if Option.isNone command then
+                        if retries > 0 then
```

Comment:
I don't quite understand why we need the retries. `tryWhere` already has retries built-in.